scripts/benchmark/sampling/sampling_visualization.py

```diff
@@ -0,0 +1,235 @@
+
+def get_middle_of_bins(bin_edges: np.ndarray) -> np.ndarray:
+    return 0.5 * (bin_edges[:-1] + bin_edges[1:])
+
+
+def compare_histograms(
+    data: pd.DataFrame,
+    sample: pd.DataFrame,
+    metric: str,
+    bins: Union[int, str, List[float]],
+) -> None:
+    fig = go.Figure()
+
+    values, bin_edges = np.histogram(data[metric].to_numpy(), bins)
+
+    fig.add_bar(
+        x=get_middle_of_bins(bin_edges),
+        y=np.round(values * len(sample) / len(data)),
+        name='Target',
+    )
+
+    values, _ = np.histogram(
+        data.merge(sample[PROJECT_COLUMN], on=PROJECT_COLUMN)[metric].to_numpy(),
+        bin_edges,
+    )
+
+    fig.add_bar(x=get_middle_of_bins(bin_edges), y=values, name='Actual')
+    fig.update_layout(xaxis_title='Bin', yaxis_title='Number of projects')
+
+    st.plotly_chart(fig, use_container_width=True)
+
+
+def main():
+    st.title('Stratified Project Sampling')
+
+    left_column, right_column = st.columns(2)
+
+    with left_column:
+        file_path = Path(st.text_input('Dataset path:'))
+
+    with right_column:
+        language = st.selectbox(
+            'Language:',
+            options=Language.values(),
+            index=Language.values().index(Language.PYTHON.value),
+        )
+
+    metrics = st.multiselect('Metrics:', options=MetricName.values())
+    if not metrics:
+        st.stop()
+
+    row_data = load_data(file_path, Language(language))
+    if row_data is None:
+        st.error('Metrics not found.')
+        st.stop()
+
+    data = row_data.dropna(subset=metrics)
+
+    left_column, right_column = st.columns(2)
+
+    with left_column:
+        st.metric('Total number of projects:', len(row_data), help='The number of projects in the dataset.')
+
+    with right_column:
+        st.metric(
+            'Actual number of projects:',
+            len(data),
+            delta=len(data) - len(row_data),
+            help='The number of projects in which all the specified metrics are present.',
+        )
+
+    st.header('Histograms')
+
+    metric_bins = {}
+    for metric in metrics:
+        st.subheader(metric)
+
+        metric_bins[metric] = get_bins(metric)
+        values, bin_edges = np.histogram(data[metric], bins=metric_bins[metric])
+
+        number_of_bins = len(bin_edges) - 1
+        number_of_non_empty_bins = len([value for value in values if value != 0])
+
+        left_column, right_column = st.columns(2)
+
+        with left_column:
+            st.metric('Number of bins:', number_of_bins)
+
+        with right_column:
+            st.metric(
+                'Number of non-empty bins:',
+                number_of_non_empty_bins,
+                delta=number_of_non_empty_bins - number_of_bins,
+            )
+
+        with st.expander('Histogram:'):
+            fig = px.bar(x=get_middle_of_bins(bin_edges), y=values)
+            fig.update_layout(xaxis_title='Bin', yaxis_title='Number of projects')
+            st.plotly_chart(fig, use_container_width=True)
```

Comment:
Currently, you have the too long main function, probably you can split it into several ones, e.g. you can move this code into a function